Source/Synthesis/wavetable_osc.h

```diff
@@ -0,0 +1,141 @@
+#pragma once
+#ifndef DSY_WAVETABLEOSC_H
+#define DSY_WAVETABLEOSC_H
+
+#include "Utility/wavetables.h"
+
+
+namespace daisysp
+{
+/**
+ * @brief Wave table oscillator
+ *
+ * Based on implementation from STK (by Perry R. Cook and Gary P. Scavone, 1995--2021)
+ */
+class WavetableOsc
+{
+  public:
+    enum Waveform
+    {
+        WAVE_SIN,
+        WAVE_TRI,
+        WAVE_SAW,
+        WAVE_SQUARE,
+        WAVE_LAST,
+    };
+
+    void Init(float sample_rate_)
+    {
+        sample_rate   = sample_rate_;
+        sr_resiprocal = 1 / sample_rate;
+        SetWaveform(WAVE_SQUARE);
+        SetFreq(440);
+        SetAmp(0.7f);
+    }
+
+    void SetWaveform(const Waveform wf)
+    {
+        switch(wf)
+        {
+            case Waveform::WAVE_SIN: SetWaveTable(&Tables::Sine); break;
+            case Waveform::WAVE_TRI: SetWaveTable(&Tables::Tri); break;
+            case Waveform::WAVE_SAW: SetWaveTable(&Tables::Saw); break;
+            case Waveform::WAVE_SQUARE: SetWaveTable(&Tables::Square); break;
+            default: SetWaveTable(&Tables::Sine);
```

Comment:
These now throw the error(s)
`'template<class T, void (* fft_func)(int, T*, T*)> class daisysp::Tables' used without template arguments` and so on.